node/node_explorer.go

```diff
@@ -148,3 +149,22 @@ func (node *Node) commitBlockForExplorer(block *types.Block) {
 		node.Consensus.PbftLog.DeleteMessagesLessThan(curNum - 100)
 	}
 }
+
+// CommitCommittee commits committee with shard id and epoch to explorer service.
+func (node *Node) CommitCommittee() {
+	events := make(chan core.ChainHeadEvent)
+	node.Blockchain().SubscribeChainHeadEvent(events)
+	for event := range events {
+		curBlock := event.Block
+		state, err := node.Blockchain().ReadShardState(curBlock.Epoch())
+		if err != nil {
+			utils.Logger().Error().Err(err).Msg("[Explorer] Error reading shard state")
+			continue
+		}
+		for _, committee := range state {
+			if committee.ShardID == curBlock.ShardID() {
+				explorer.GetStorageInstance(node.SelfPeer.IP, node.SelfPeer.Port, false).DumpCommittee(curBlock.ShardID(), curBlock.Epoch().Uint64(), committee)
```

Comment:
Errors should be checked and logged if needed.  If done here, it eliminates need for logging from within `DumpCommittee`.